pkg/store/bucket.go

```diff
@@ -716,6 +719,25 @@ func blockSeries(
 			return s.lset[i].Name < s.lset[j].Name
 		})
 
+		validTombstones := make([]tombstone.Tombstone, 0)
+		for _, ts := range tombstones {
+			tsMatchers, err := parser.ParseMetricSelector(ts.Matchers)
+			if err != nil {
+				return nil, nil, err
+			}
+			tsPostings, err := indexr.ExpandedPostings(tsMatchers)
```

Comment:
While this is correct, there might be more optimal solution. 


Instead of doing matchers to postings lookup we could just match labels that we have vs matchers that tombstones specifies. Thanks to this we don't need fetch potentially thousands of unrelated postings during this lookup.

Reply:
Let's say we query for `cluster=us1`


Deleted `cluster=~"us.*"`

so we here we have in `lset` cluster=us1 but still we  (maybe there are 100000 us cluster) would fetch 10000 of those postings here.

Reply:
Can we encapsulate some of this in Tombstone methods? (: